### DISCUSSION TITLE
Why the same training runs faster in WSL than in PowerShell?

### DISCUSSION THREAD

**Discussion by hopezh** [rasbt/machine-learning-notes#40] [Q&A]
(The following was also posted on [Reddit](https://www.reddit.com/r/deeplearning/comments/15sg4tx/why_the_same_training_runs_faster_in_wsl_than_in/). But I think I should ask experts here directly) 

I ran the [vgg16-cifar10](https://github.com/rasbt/machine-learning-notes/blob/main/benchmark/pytorch-m1-gpu/vgg16-cifar10.py) script in both PowerShell and wsl, consecutively,

However, the script ran in PowerShell spent three times longer to finish the training as compared to that ran in WSL, as shown in the images below.

1. script ran in powershell
![2023-08-16-11-14-47 powershell vgg16](https://github.com/rasbt/machine-learning-notes/assets/23435810/5ffe4c1c-6066-4496-ad26-20dac329a43e)

2. script ran in WSL
![2023-08-16-11-14-47 wsl vgg16](https://github.com/rasbt/machine-learning-notes/assets/23435810/dbbeee00-e429-4fe1-b919-9f70c2696c30)

My question:
**_Why the same training script ran faster in WSL than in PowerShell when both were accessing the same GPU?_**

My computer config:
OS: Windows 11 Pro for workstation
CPU: Intel(R) Xeon(R) Gold 5222 CPU @ 3.80GHz 3.79 GHz (2 processors)
RAM: 128 GB
Graphic card: NVIDIA Quadro RTX 4000 (8GB)
PowerShell: v7.3.6
WSL2: v[1.2.5.0](https://1.2.5.0/), with Ubuntu-22.04.2 LTS installed

**Comment by rasbt** [comment]
That's a good question. Tbh I have no idea since I never used PowerShell and wsl. (I am a macOS & Linux user.)

Maybe one of them has multiprocessing limitations and the data loading is slower so that it becomes a bottleneck. But I am honestly just guessing here.
